### PR TITLE
Feature/36 improve response types for grpc support

### LINKED ISSUE #36: Improve response types for grpc support
I wanna add some grpc attribute to make response grpc (Code-First) compatible.

### PR DESCRIPTION
**Pull request by Reza-Noei**
ITNOA

Please review this PR 🌹

fixes #36

### CODE REVIEW

File: Source/BSN.Commons.Orm.EntityFramework/BSN.Commons.Orm.EntityFramework.csproj [hunk 1]
```diff
@@ -1,80 +1,81 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (soroshsabz) [inline Source/BSN.Commons.Orm.EntityFramework/BSN.Commons.Orm.EntityFramework.csproj:1]: Why these changes?

File: Source/BSN.Commons.Orm.EntityFrameworkCore/Extensions/IQueryableExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.EntityFrameworkCore;
```
Comment (soroshsabz) [inline Source/BSN.Commons.Orm.EntityFrameworkCore/Extensions/IQueryableExtensions.cs:1]: These file is redundant, please move all of them to main  BSN.Commons

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs [hunk 3]
```diff
@@ -1,13 +0,0 @@
-﻿namespace BSN.Commons.Responses
```
Comment (soroshsabz) [inline Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs:1]: For backward compatibility issues, please do not remove these classes

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs [hunk 4]
```diff
@@ -1,17 +0,0 @@
-﻿namespace BSN.Commons.Responses
```
Comment (soroshsabz) [inline Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBaseWithPagination.cs:1]: For backward compatibility issues, please do not remove these classes

**Review by soroshsabz — CHANGES_REQUESTED** [review]
@Reza-Noei Please consider the notes

thanks

File: Source/BSN.Commons.Orm.EntityFrameworkCore/Extensions/IQueryableExtensions.cs (hunk 2 above)
Comment (Reza-Noei) [reply Source/BSN.Commons.Orm.EntityFrameworkCore/Extensions/IQueryableExtensions.cs:1]: ToListAsync is Framework dependent. It has different implementation in .EntityFramework and .EFCore.

I can move Paginate extension to BSN.Commons but its ugly.

File: Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs (hunk 3 above)
Comment (Reza-Noei) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs:1]: It's a breaking change. we cant do this.

Comment (soroshsabz) [reply Source/BSN.Commons.PresentationInfrastructure/Responses/GenericResponseBase.cs:1]: Where is contructor?

**Review by soroshsabz — APPROVED** [review]
